### PR TITLE
Ekf2 accuracy improvement

### PR DESCRIPTION
**Pull request by priseborough**
See https://github.com/PX4/ecl/pull/117 for details

### CODE REVIEW

File: msg/ekf2_replay.msg [hunk 1]
```diff
@@ -37,3 +37,10 @@ float32[2] flow_pixel_integral		# integrated optical flow rate around x and y ax
 float32[2] flow_gyro_integral		# integrated gyro rate around x and y axes (rad)
 uint32 flow_time_integral		# integration timespan (usec)
 uint8 flow_quality			# Quality of accumulated optical flow data (0 - 255)
+
+# vehicle status
+uint8 vehicle_status_mask		# mask of vehcle status - bit positions defined as:
+# 0 - true if the vehicle is landed
+# 1 - true if the landed status has been updated on this frame
+# 2 - true if the actuators/motors are armed
+# 3 - true if the armed status has been updated on this frame
```
Comment (julianoes) [inline msg/ekf2_replay.msg]: I don't quite understand why this is needed. Can't you re-use the existing logged messages `LAND.Landed` and `STAT.ArmS`?

Comment (priseborough) [inline msg/ekf2_replay.msg]: I wasn't aware they were available when replay logging was active

Comment (priseborough) [inline msg/ekf2_replay.msg]: I will pull the logging patches out of the accuracy improvement PR and rework them them as a separate PR

**Comment by priseborough** [comment]
Note https://github.com/PX4/Firmware/pull/4484/commits/ba5972e6d9d0e330fd0a67c84a0035154f5f35e2
removes the estimators reliance on the arm status message which wasn't being included in the replay logging. The interface is now cleaner with a single external boolean determining if the estimator thinks it is airborne or not.
Flight testing and replay has verified that the change to and from in-air is being detected correctly by ekf2.

File: msg/ekf2_replay.msg (hunk 1 above)
Comment (julianoes) [inline msg/ekf2_replay.msg]: Aha, the catch is that they are only available on Snapdragon because there **all** are available.

**Comment by julianoes** [comment]
This crashes on Snapdragon. Looking into it.

**Comment by julianoes** [comment]
@tumbili suggested that it might be happening in:
https://github.com/PX4/ecl/blob/e91a934/EKF/estimator_interface.cpp#L113

If `_imu_buffer` is bigger (100 instead of 30 bytes), it doesn't crash. Still, the root cause needs to be fixed.

**Comment by julianoes** [comment]
It seems that the crashes on Snapdragon were related to stack size. I've raised the stack size used for QURT but I suggest to check on NuttX as well.

**Comment by LorenzMeier** [comment]
@tumbili Can you re-test the stack size with this on NuttX?

**Comment by LorenzMeier** [comment]
Rebased and applied, thanks!

**Comment by julianoes** [comment]
Test flown on Snapdragon 2x 6mins of POSCTL with fairly good performance (and no major problems).

**Pull request closed without merging**